### DISCUSSION TITLE
深度桌面时间显示能不能把那个年份去掉啊

### DISCUSSION THREAD

**Discussion by bigben446** [linuxdeepin/developer-center#4075] [Features Request & Ideas | 特性请求 & 头脑风暴]
年份特别占地方，和月份日期挤在一起，谁不知道今年年份是多少啊。那个地方加一个星期几倒是可以啊。

理想中的时间日期极简模式，两行，一行时间，一行星期和日期，星期只要写汉字就行，日期只要写日月，不要年份：

时间：8:44

星期.日月：二.4/11

https://bbs.deepin.org/zh/post/254923

**Comment by BLumia** [comment]
提供一个配置项，然后深度定制里加个配置它的地方？不同用户的习惯不一样，比如我倾向于有年份且完全不需要星期的信息。KDE 这种可配置的方案应该好一些但并不是所有用户都需要这种程度的定制性。

**Reply by bigben446** [reply]
对于有强迫症的用户来说，能像kde用户一样自定义解决问题最好。要不然怎么看怎么不顺眼，你看我3年之前发了一个类似的帖子，3年了还是没习惯，忍不住再发了一个同样内容的帖子。关键是这个问题，kde已经解决了，参考代码移植过来就是了。

**Reply by bigben446** [reply]
不需要定制性，那就使用默认参数显示就是了，不矛盾。

**Reply by BLumia** [reply]
实现这个从来都不是难事，关注点反而应该是如何恰当提供这种定制或者配置的能力。对于完全不熟悉这方面信息的人，ta可能不知道 ddd MM 表示什么意思。

cc @meiyixiang

**Reply by BLumia** [reply]
> 那就使用默认参数显示就是了

那我绝对会反对将你所倾向的显示方式作为默认显示方式。

**Reply by bigben446** [reply]
> > 那就使用默认参数显示就是了
> 
> 那我绝对会反对将你所倾向的显示方式作为默认显示方式。

不矛盾，就以现有的显示方式作为默认。我需要的是一个高级用户可定制的特性。

**Reply by bigben446** [reply]
> 实现这个从来都不是难事，关注点反而应该是如何恰当提供这种定制或者配置的能力。对于完全不熟悉这方面信息的人，ta可能不知道 ddd MM 表示什么意思。
> 
> cc @meiyixiang

就像那个深度标题栏高度一样，可以自己定制高度，想要多高就多高，也不至于那么多用户抱怨丑丑的大额头了

**Reply by BLumia** [reply]
那不就是我最初回复的方案吗，dock 的时间插件加一个配置项，配置入口提供在深度定制里 :joy:

**Reply by BLumia** [reply]
有兴趣 PR 吗？这个其实相当好改。

**Reply by bigben446** [reply]
> 有兴趣 PR 吗？这个其实相当好改。

啥是PR？有兴趣啊，有很强的需求和动力，但是不知道怎么改

**Reply by BLumia** [reply]
Pull Requests，是 GitHub 发起代码贡献的途径，具体步骤可以先看看 GitHub 文档熟悉一下。不过在发起贡献之前，你可以试着修改代码，实现一下这个需求。

这个逻辑位于 dde-dock 项目的时间插件里，目前时间和日期格式是固定的，需要修改代码使之可从配置里读取显示格式。

如果你有兴趣的话也欢迎加入我们的开发者实时交流 [Matrix](https://wiki.deepin.org/Matrix) 群，然后沟通会更方便一些 :)

**Comment by bigben446** [comment]
相关的讨论

深度桌面时间显示能不能把那个年份去掉啊
https://bbs.deepin.org/zh/post/254923

deepin任务栏时间能否去掉年份
https://bbs.deepin.org/post/203987

任务栏侧边显示后时间显示文字非常小
https://bbs.deepin.org/post/224769

**Comment by bigben446** [comment]
关键是kde已经早就实现这个功能了，高度自定义化时间显示，帖子的6楼有一个效果图就是这个
https://bbs.deepin.org/zh/post/254923

显示效果图片的链接
https://storage.deepin.org/thread/202304111157125746_Screenshot_20230411_115705.png

**Comment by meiyixiang** [comment]
功能实现很简单，不过这里做起来会需要考虑很多自由定义的方式包括UI布局等多方面问题，这个问题在需求池里记录了有，但是现在不会马上做，等系统现有主要功能和使用方面的问题完成后会考虑排期去实现

**Reply by bigben446** [reply]
其实就是一个日期格式那边增加一个可定制选项

**Reply by meiyixiang** [reply]
我理解你说的意思，可以选择不展示年份，只展示日期和时间格式，或者选择可以展示星期几。

**Comment by meiyixiang** [comment]
如果你可以有现成的方案，可根据现有DDE布局样式像我们提交PR，没问题我们可以和入。

**Reply by bigben446** [reply]
好的，我去看看，哪儿有相关的资料可以参考

**Reply by meiyixiang** [reply]
我转到我们内部开发群里了，稍后会有相关开发回复你。

**Reply by bigben446** [reply]
> 我转到我们内部开发群里了，稍后会有相关开发回复你。

好的，谢谢

**Reply by meiyixiang** [reply]
Pull Requests，是 GitHub 发起代码贡献的途径，具体步骤可以先看看 GitHub 文档熟悉一下。不过在发起贡献之前，你可以试着修改代码，实现一下这个需求。

这个逻辑位于 dde-dock 项目的时间插件里，目前时间和日期格式是固定的，需要修改代码使之可从配置里读取显示格式。

如果你有兴趣的话也欢迎加入我们的开发者实时交流 [Matrix](https://wiki.deepin.org/Matrix) 群，然后沟通会更方便一些 :)

**Reply by meiyixiang** [reply]
上面我们同事描述的方式，你可以看看有问题可以加入交流群。

**Reply by bigben446** [reply]
> Pull Requests，是 GitHub 发起代码贡献的途径，具体步骤可以先看看 GitHub 文档熟悉一下。不过在发起贡献之前，你可以试着修改代码，实现一下这个需求。
> 
> 这个逻辑位于 dde-dock 项目的时间插件里，目前时间和日期格式是固定的，需要修改代码使之可从配置里读取显示格式。
> 
> 如果你有兴趣的话也欢迎加入我们的开发者实时交流 [Matrix](https://wiki.deepin.org/Matrix) 群，然后沟通会更方便一些 :)

现在开发是基于v23还是v20.9？

**Reply by BLumia** [reply]
主干是 23，你自己在用 20.9 的话基于 20.9 的代码改也可以（我们不会往主干合入），要改的东西差别不大。

**Reply by bigben446** [reply]
> 主干是 23，你自己在用 20.9 的话基于 20.9 的代码改也可以（我们不会往主干合入），要改的东西差别不大。

好的，谢谢

**Comment by Aninana** [comment]
这个建议好，就像Windows小任务栏那样显示。
